homeassistant/components/tod/binary_sensor.py

```diff
@@ -174,6 +174,16 @@ def _calculate_initial_boudary_time(self):
 
         self._time_before = before_event_date
 
+        # We are calculating the _time_after value assuming that it will happen today
+        # But it's not always truth. E.g. after 23:00, before 12:00, and now is 10:00
+        # Here we check if _time_after was yesterday and _time_before will be today
+        if (
+            self._time_after > self.current_datetime
+            and self._time_before > self.current_datetime + timedelta(days=1)
+        ):
```

Comment:
> Here we check if _time_after was yesterday

Aren't you checking here if time_after is later/greater than current_datetime?

> and _time_before will be today

And aren't you checking here if _time_before is later than tomorrow (current_datetime + 1 day)?
Or am I misunderstanding something? 😅

Reply:
Yes, you're right.
But above `_time_after` was set for current day.
E.g. if we set `after: 23:00`, `before: 12:00` and now is 10:00, in code above `_time_after` set for 23:00 current day and `_time_before` for 12:00 next day. To fix this i decrease these values.

Reply:
Okay now I got it ✌
Then I would recommend to replace line 179 with the following:
```python
# If _time_before and _time_after are ahead of current_datetime:
# _time_before is set to 12:00 next day
# _time_after is set to 23:00 today
# current_datetime is set to 10:00 today
```